### DISCUSSION TITLE
Startup Behaviour with multiple files

### DISCUSSION THREAD

**Discussion by luctius** [helix-editor/helix#444] [Ideas] [answered]
Right now, when starting hx with multiple files `hx *.rs` for example, a split view will be created for each file. I think we should limit this somehow. My suggestion is to create a config setting with which we can limit this, and personally I would set it default to two. Meaning files past the first two will be opened in the buffer but will not get their own view, during startup.

I'm willing to do this; but I first wanted your input on this behaviour and it's possible alternatives.

**Comment by pickfire** [comment, marked as answer]
At least for me, I don't think it will be useful to start multiple files as splits, because if user specified 10 files means there will be 10 splits? So I think those other files should be hidden. Not sure what others think. One reason is probably because my terminal is usually 80 width.

**Reply by luctius** [reply]
> because if user specified 10 files means there will be 10 splits? 

That is the current behaviour when started with multiple files. My suggestion thus is to limit it. I would say to 2, but 1 (so only the main view) is also fine with me.

**Reply by pickfire** [reply]
On 80x24 terminal even a vertical split will look ugly. I think only diff should split by default. Also, we don't know if user want a horizontal split or vertical split.

**Reply by pickfire** [reply]
What's your use case? Why do you specify multiple files to start?

**Reply by luctius** [reply]
I like to specify the files I am initially going to work with from the commandline. As an example; my work with the switch_case command. There I started helix with `hx helix-term/src/{command,keymap}.rs`. In that case the editor starts with 1 additional view.

But to be clear; I also don't mind if helix would start with only the initial view, however that is not it's current behaviour. At the moment the view is split for each additional file given from the commandline; ad infinitum. However making this behaviour configurable seems to me the best option.

**Reply by pickfire** [reply]
Ah, seemed like current it already starts with a split. Did you checked master?

**Reply by luctius** [reply]
Forgive me I'm wrong; but I think we are talking past each other :-).

The **Current** behaviour (on master)  when specifying multiple files as starting arguments is:

- With 1 file, 1 viewport.
- With 2 files, 2 viewports.
- with 8 files, 8 viewports.
- With 100 files -> crash.

My suggestion is to limit it in some way.
Either force only 1 viewport (which to me seems to be your preference), or create a config setting somewhere which states the maximum number of viewports created during startup.

**Reply by pickfire** [reply]
Sure, 8 viewports is bad. I think like you said, limiting it to as few viewports as possible is the best option.

**Reply by sudormrfbin** [reply]
Adding a config option seems overkill, I think starting with a single viewport set to the first file and loading the rest into buffers is good enough. We can maybe show a message like "Loaded n files" in the command line.

**Comment by aral** [comment]
Just a quick note for anyone reading this at a later date: the behaviour seems to have been changed so that when opening multiple files, only one shows but all of them are available in your open buffers (<kbd>space</kbd><kbd>b</kbd> to view them). And this works really well :)
